libp2p/builders.nim

```diff
@@ -166,7 +166,9 @@ proc build*(b: SwitchBuilder): Switch
 
   let
     identify = Identify.new(peerInfo)
-    connManager = ConnManager.new(b.maxConnsPerPeer, b.maxConnections, b.maxIn, b.maxOut)
+    maxIn = if b.maxIn >= 0: b.maxIn else: b.maxConnections
```

Comment:
What happens if `maxIn/maxOut > maxConnections`?

Reply:
Without forceDial, you'll always be capped by the lowest value. With forceDial, you can go over maxConnections